holoviews/operation/datashader.py

```diff
@@ -0,0 +1,208 @@
+
+    x_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
+    y_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
+    @classmethod
+    def get_agg_data(cls, obj, category=None):
+        paths = []
+        kdims = obj.kdims
+        vdims = obj.vdims
+        x, y = obj.dimensions(label=True)[:2]
+        if isinstance(obj, Path):
+            line = True
+            for p in obj.data:
+                df = pd.DataFrame(p, columns=obj.dimensions('key', True))
+                if isinstance(obj, Contours) and obj.vdims and obj.level:
+                    df[obj.vdims[0].name] = p.level
+                paths.append(df)
+        elif isinstance(obj, CompositeOverlay):
+            for key, el in obj.data.items():
+                x, y, element, line = cls.get_agg_data(el)
+                df = element.dframe()
+                if isinstance(obj, NdOverlay):
+                    df = df.assign(**dict(zip(obj.dimensions('key', True), key)))
+                paths.append(df)
+            kdims += element.kdims
+            vdims = element.vdims
+        elif isinstance(obj, Element):
+            line = isinstance(obj, Curve)
+            paths.append(obj.dframe())
+        if line:
+            empty = paths[0][:1].copy()
+            empty.loc[0, :] = (np.NaN,) * empty.shape[1]
+            paths = [elem for path in paths for elem in (path, empty)][:-1]
+        df = pd.concat(paths).reset_index(drop=True)
+        if category and df[category].dtype.name != 'category':
+            df[category] = df[category].astype('category')
+        return x, y, Dataset(df, kdims=kdims, vdims=vdims), line
+
+
+    def _process(self, element, key=None):
+        agg_fn = self.p.aggregator
+        category = agg_fn.column if isinstance(agg_fn, ds.count_cat) else None
+        x, y, data, line = self.get_agg_data(element, category)
+
+        if self.p.x_range:
+            xstart, xend = self.p.x_range
+        else:
+            xstart, xend = data.range(x)
+        if self.p.y_range:
+            ystart, yend = self.p.y_range
+        else:
+            ystart, yend = data.range(y)
+
+        # Compute highest allowed sampling density
+        width, height = self.p.width, self.p.height
+        if self.x_sampling:
+            x_range = xend - xstart
+            width = int(min([(x_range/self.p.x_sampling), width]))
+        if self.y_sampling:
+            y_range = yend - ystart
+            height = int(min([(y_range/self.p.y_sampling), height]))
+
+        cvs = ds.Canvas(plot_width=width, plot_height=height,
+                        x_range=(xstart, xend), y_range=(ystart, yend))
+
+        if line:
+            return cvs.line(data, x, y, self.p.aggregator)
+        else:
+            return cvs.points(data, x, y, self.p.aggregator)
+            
```

Comment:
Maybe instead of a boolean `line`, this could be handled more generally by setting `glyph=cvs.points` or `glyph=cvs.line` above, then doing `return getattr(cvs,glyph)(data, x, y, self.p.aggregator)` here?